docs/core/PORTING.md

```diff
@@ -380,16 +420,27 @@ SECTIONS
         __uvisor_bss_boxes_end = .;
 
         /************************** ARMv7-M MPU only **************************/
-        __uvisor_bss_end_padding_max = (2 << (LOG2CEIL(__uvisor_bss_end - ORIGIN(m_data)) - 1)) / 8;
-        . = MIN(
-            __uvisor_bss_end_padding_max * (((__uvisor_bss_end - ORIGIN(m_data)) / __uvisor_bss_end_padding_max) +
-            MIN((__uvisor_bss_end - ORIGIN(m_data)) % __uvisor_bss_end_padding_max, 1)) - __UVISOR_SRAM_OFFSET,
-            ORIGIN(m_data) + LENGTH(m_data)
-        );
+        . = ALIGN((1 << LOG2CEIL(LENGTH(m_data))) / 8);
+        /********************** Other MPU architectures ***********************/
+        . = ALIGN(32);
+        /**********************************************************************/
+        __uvisor_bss_end = .;
+    } > m_data
+
+    /* Ensure that the page heap is put right after the uVisor BSS section in
+     * SRAM. */
+    /* Heap space for the page allocator */
+    .page_heap (NOLOAD) :
+    {
+        . = ALIGN(32);
+        __uvisor_page_start = .;
+        KEEP(*(.keep.uvisor.page_heap))
+        /************************** ARMv7-M MPU only **************************/
+        . = ALIGN((1 << LOG2CEIL(LENGTH(m_data))) / 8);
```

Comment:
This is fine, except this may fail, if the `__uvisor_page_size > (1 << LOG2CEIL(LENGTH(m_data))) / 8`. I'm not sure if that should be handled in the linkerscript, or if we just do not allow this page size.